deeposlandia/kerastrain.py

```diff
@@ -0,0 +1,226 @@
+                        default="./data",
+                        help="Relative path towards data directory")
+    return parser
+
+def add_hyperparameters(parser):
+    """Add hyperparameter arguments from the command line
+
+    Parameters
+    ----------
+    parser : argparse.ArgumentParser
+        Input parser
+
+    Returns
+    -------
+    argparse.ArgumentParser
+        Modified parser, with additional arguments
+    """
+    parser.add_argument('-b', '--batch-size',
+                        type=int,
+                        default=20,
+                        help=("Number of images that must be contained "
+                              "into a single batch"))
+    parser.add_argument('-d', '--dropout',
+                        type=float,
+                        default=1.0,
+                        help=("Percentage of dropped out neurons "
+                              "during training"))
+    parser.add_argument('-e', '--nb-epochs',
+                        type=int,
+                        default=0,
+                        help=("Number of training epochs (one epoch means "
+                              "scanning each training image once)"))
+    parser.add_argument('-l', '--learning-rate',
+                        nargs="+",
+                        default=[0.001, 300, 0.95],
+                        type=float,
+                        help=("List of learning rate components (starting LR, "
+                              "decay steps and decay rate)"))
+    parser.add_argument('-s', '--image-size',
+                        default=256,
+                        type=int,
+                        help=("Desired size of images (width = height)"))
+    return parser
+
+def add_training_arguments(parser):
+    """Add training-specific arguments from the command line
+
+    Parameters
+    ----------
+    parser : argparse.ArgumentParser
+        Input parser
+
+    Returns
+    -------
+    argparse.ArgumentParser
+        Modified parser, with additional arguments
+    """
+    parser.add_argument('-ii', '--nb-testing-image',
+                        type=int,
+                        default=5000,
+                        help=("Number of training images"))
+    parser.add_argument('-it', '--nb-training-image',
+                        type=int,
+                        default=18000,
+                        help=("Number of training images"))
+    parser.add_argument('-iv', '--nb-validation-image',
+                        type=int,
+                        default=2000,
+                        help=("Number of validation images"))
+    return parser
+
+if __name__=='__main__':
+    """Main method: run a convolutional neural network using Keras API
+    """
+
+    # Parse command-line arguments
+    parser = argparse.ArgumentParser(description=("Convolutional Neural Netw"
+                                                  "ork on street-scene images"))
+    parser = add_instance_arguments(parser)
```

Comment:
split the arg parser into different functions is a good idea! Well done.

Reply:
Thanks. :)
Actually that was a real mess, we need to be more organized. I hope it is good enough!